openmc/filter.py

```diff
@@ -77,6 +77,23 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self == other
 
+    def __gt__(self, other):
+        if self.type != other.type:
+            if self.type in _FILTER_TYPES and other.type in _FILTER_TYPES:
+                delta = _FILTER_TYPES.index(self.type) - \
+                        _FILTER_TYPES.index(other.type)
+                return True if delta > 0 else False
+            else:
+                return False
+        else:
+            if 'energy' in self.type and 'energy' in other.type:
```

Comment:
Could you add a comment that describes the logic of comparing whether one energy filter is greater than another?